src/nnet3/nnet-example-utils.cc

```diff
@@ -198,6 +198,54 @@ void ShiftExampleTimes(int32 t_offset,
     }
   }
 }
+void GetComputationRequest(const Nnet &nnet,
+                           const NnetExample &eg,
+                           bool need_model_derivative,
+                           bool store_component_stats,
+                           bool use_xent_regularization,
+                           bool use_xent_derivative,
+                           ComputationRequest *request) {
```

Comment:
Perhaps you can combine with the previous function and give the two new arguments at the end with the default as false.